utils/UUID_gen.hh

```diff
@@ -289,13 +289,18 @@ public:
      * <b>Warning:</b> this method should only be used for querying as this
      * doesn't at all guarantee the uniqueness of the resulting UUID.
      */
-    static UUID min_time_UUID(milliseconds timestamp = milliseconds{0})
+    template <std::intmax_t N, std::intmax_t D>
+    static UUID min_time_UUID(std::chrono::duration<int64_t, std::ratio<N, D>> timestamp)
```

Comment:
It's enough to use decimicroseconds as the type here, if you pass milliseconds or microseconds it will get auto-converted (and if you pass a higher resolution type it will be rejected).

Note it's odd to pass a duration as a timestamp, the right type is a time_point, but that's a different matter.